ugs-pendant/src/main/java/com/willwinder/universalgcodesender/pendantui/v1/controllers/MachineController.java

```diff
@@ -0,0 +1,238 @@
+package com.willwinder.universalgcodesender.pendantui.v1.controllers;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.willwinder.universalgcodesender.IController;
+import com.willwinder.universalgcodesender.connection.ConnectionDriver;
+import com.willwinder.universalgcodesender.connection.ConnectionFactory;
+import com.willwinder.universalgcodesender.model.Axis;
+import com.willwinder.universalgcodesender.model.BackendAPI;
+import com.willwinder.universalgcodesender.model.BaudRateEnum;
+import com.willwinder.universalgcodesender.pendantui.v1.model.GcodeCommands;
+import com.willwinder.universalgcodesender.services.JogService;
+import com.willwinder.universalgcodesender.utils.FirmwareUtils;
+import com.willwinder.universalgcodesender.utils.Settings;
+import com.willwinder.universalgcodesender.utils.SettingsFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.BufferedReader;
+import java.io.StringReader;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Path("/v1/machine")
+public class MachineController {
+
+    @Inject
+    private BackendAPI backendAPI;
+
+    @Inject
+    private JogService jogService;
+
+    @GET
+    @Path("connect")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response connect() throws Exception {
+        if (backendAPI.isConnected()) {
+            return Response.status(Response.Status.NOT_ACCEPTABLE).build();
```

Comment:
When you mentioned error handling in the PR comment, does that include things like setting the status messages in calls like this?

Maybe `backendAPI.connect` should throw a usable exception message so that can be used instead of checking them here as well?

Reply:
I've cleaned this up a bit using exceptions suitable for this. Actually, the error handling that I mentioned is in the GUI. But I think this needs to mature a bit before adding that.